homeassistant/components/rest/switch.py

```diff
@@ -97,8 +106,17 @@ async def async_setup_platform(
             "Missing resource or schema in configuration. "
             "Add http:// or https:// to your URL"
         )
-    except (asyncio.TimeoutError, aiohttp.ClientError):
-        _LOGGER.error("No route to resource/endpoint: %s", resource)
+    except (asyncio.TimeoutError, aiohttp.ClientError) as exc:
+        if config.get(CONF_RESOURCE) is not None:
+            rendered_resource = config[CONF_RESOURCE]
+        else:
+            resource: template.Template = config[CONF_RESOURCE_TEMPLATE]
+            resource.hass = hass
+            rendered_resource = resource.async_render_with_possible_json_value(
+                resource, "http://127.0.0.1"
+            )
+        _LOGGER.error("No route to resource/endpoint: %s", rendered_resource)
+        raise PlatformNotReady() from exc
```

Comment:
This is a bugfix that should be standalone, in a separate preliminary PR.

Reply:
agreed, addressing it in #90627. Then I will change the code is this P.R.